erpnext/accounts/doctype/bank_account/bank_account.json

```diff
@@ -208,8 +208,54 @@
    "label": "Disabled"
   }
  ],
- "links": [],
- "modified": "2024-03-27 13:06:37.049542",
+ "links": [
+  {
+   "group": "Transactions",
+   "link_doctype": "Payment Request",
+   "link_fieldname": "bank_account"
+  },
+  {
+   "group": "Transactions",
+   "link_doctype": "Payment Order",
+   "link_fieldname": "bank_account"
+  },
+  {
+   "group": "Transactions",
+   "link_doctype": "Bank Guarantee",
+   "link_fieldname": "bank_account"
+  },
+  {
+   "group": "Transactions",
+   "link_doctype": "Payroll Entry",
```

Comment:
Should be part of HRMS App.

Installation of Banking app failing because of this:
```
Installing banking...
Updating DocTypes for banking       : [========================================] 100%
Installing Banking Customizations ...
An error occurred while installing banking: Document Links Row #4: Invalid doctype or fieldname.('DocType', 'Payroll Entry')
Traceback with variables (most recent call last):
  File "apps/frappe/frappe/commands/site.py", line 452, in install_app
    _install_app(app, verbose=context.verbose, force=force)
      context = {'sites': ['lush.localhost'], 'force': False, 'verbose': False, 'profile': False}
      apps = ('banking',)
      force = False
      _install_app = <function install_app at 0x7f6e9e4fccc0>
      filelock = <function filelock at 0x7f6e9e4ea480>
      exit_code = 0
      site = 'lush.localhost'
      app = 'banking'
      err = InvalidFieldNameError("Document Links Row #4: Invalid doctype or fieldname.('DocType', 'Payroll Entry')")
  File "apps/frappe/frappe/installer.py", line 322, in install_app
    frappe.get_attr(after_install)()
      name = 'banking'
      verbose = False
      set_as_patched = True
      force = False
      sync_jobs = <function sync_jobs at 0x7f6e9e557920>
      sync_for = <function sync_for at 0x7f6e9e568900>
      sync_customizations = <function sync_customizations at 0x7f6ea0b07c40>
      sync_fixtures = <function sync_fixtures at 0x7f6e9e568c20>
      app_hooks = {'after_install': ['banking.install.after_install'], 'app_description': ['Banking Integration by ALYF GmbH'], 'app_email': ['hallo@alyf.de'], 'app_include_css': ['bank_reconciliation_beta.bundle.css'], 'app_license': ['GPLv3'], 'app_name': ['banking'], 'app_publisher': ['ALYF GmbH'], 'app_title': ['ALYF Banking'], 'app_version': ['15.6.1'], 'before_tests': ['banking.utils.before_tests'], 'doc_events': {'Bank Transaction': {'on_update_after_submit': ['banking.overrides.bank_transaction.on_update_after_submit']}}, 'get_matching_queries': ['banking.klarna_kosma_integration.doctype.bank_reconciliation_tool_beta.bank_reconciliation_tool_beta.get_matching_queries'], 'kosma_custom_fields': {'Bank Account': [{'owner': 'Administrator', 'fieldname': 'kosma_account_id', 'label': 'Kosma Account ID', 'fieldtype': 'Data', 'insert_after': 'mask', 'read_only': 1, 'translatable': 0}]}, 'kosma_property_setters': {'Bank Account': [{'fieldname': 'last_integration_date', 'property': 'read_only', 'value': 1...
      installed_apps = ['frappe', 'erpnext', 'material_resource_planning', 'test_runner']
      after_install = 'banking.install.after_install'
  File "apps/banking/banking/install.py", line 10, in after_install
    create_custom_fields(frappe.get_hooks("kosma_custom_fields"))
  File "apps/frappe/frappe/custom/doctype/custom_field/custom_field.py", line 337, in create_custom_fields
    custom_field.save()
      custom_fields = {'Bank Account': [{'owner': 'Administrator', 'fieldname': 'kosma_account_id', 'label': 'Kosma Account ID', 'fieldtype': 'Data', 'insert_after': 'mask', 'read_only': 1, 'translatable': 0}]}
      ignore_validate = False
      update = True
      doctypes_to_update = {'Bank Account'}
      doctypes = ('Bank Account',)
      fields = [{'owner': 'Administrator', 'fieldname': 'kosma_account_id', 'label': 'Kosma Account ID', 'fieldtype': 'Data', 'insert_after': 'mask', 'read_only': 1, 'translatable': 0}]
      doctype = 'Bank Account'
      df = {'owner': 'Administrator', 'fieldname': 'kosma_account_id', 'label': 'Kosma Account ID', 'fieldtype': 'Data', 'insert_after': 'mask', 'read_only': 1, 'translatable': 0}
      field = 'Bank Account-kosma_account_id'
      custom_field = <CustomField: Bank Account-kosma_account_id>
  File "apps/frappe/frappe/model/document.py", line 337, in save
    return self._save(*args, **kwargs)
      self = <CustomField: Bank Account-kosma_account_id>
      args = ()
      kwargs = {}
  File "apps/frappe/frappe/model/document.py", line 390, in _save
    self.run_post_save_methods()
      self = <CustomField: Bank Account-kosma_account_id>
      ignore_permissions = None
      ignore_version = None
  File "apps/frappe/frappe/model/document.py", line 1128, in run_post_save_methods
    self.run_method("on_update")
      self = <CustomField: Bank Account-kosma_account_id>
  File "apps/frappe/frappe/model/document.py", line 962, in run_method
    out = Document.hook(fn)(self, *args, **kwargs)
      self = <CustomField: Bank Account-kosma_account_id>
      method = 'on_update'
      args = ()
      kwargs = {}
      fn = <function Document.run_method.<locals>.fn at 0x7f6ea1ac0f40>
  File "apps/frappe/frappe/model/document.py", line 1322, in composer
    return composed(self, method, *args, **kwargs)
      self = <CustomField: Bank Account-kosma_account_id>
      args = ()
      kwargs = {}
      hooks = [<function clear_doctype_notifications at 0x7f6e9eab20c0>, <function process_workflow_actions at 0x7f6e9daefec0>, <function attach_files_to_document at 0x7f6ea0e41f80>, <function apply at 0x7f6e9db1ad40>, <function update_due_date at 0x7f6e9db1ade0>, <function apply_permissions_for_non_standard_user_type at 0x7f6e9db2c5e0>]
      method = 'on_update'
      doc_events = {'*': {'on_update': ['frappe.desk.notifications.clear_doctype_notifications', 'frappe.workflow.doctype.workflow_action.workflow_action.process_workflow_actions', 'frappe.core.doctype.file.utils.attach_files_to_document', 'frappe.automation.doctype.assignment_rule.assignment_rule.apply', 'frappe.automation.doctype.assignment_rule.assignment_rule.update_due_date', 'frappe.core.doctype.user_type.user_type.apply_permissions_for_non_standard_user_type'], 'after_rename': ['frappe.desk.notifications.clear_doctype_notifications'], 'on_cancel': ['frappe.desk.notifications.clear_doctype_notifications', 'frappe.workflow.doctype.workflow_action.workflow_action.process_workflow_actions', 'frappe.automation.doctype.assignment_rule.assignment_rule.apply'], 'on_trash': ['frappe.desk.notifications.clear_doctype_notifications', 'frappe.workflow.doctype.workflow_action.workflow_action.process_workflow_actions'], 'on_update_after_submit': ['frappe.workflow.doctype.workflow_action.workflow_action.process_w...
      handler = 'frappe.core.doctype.user_type.user_type.apply_permissions_for_non_standard_user_type'
      composed = <function Document.hook.<locals>.compose.<locals>.runner at 0x7f6e9da3b920>
      compose = <function Document.hook.<locals>.compose at 0x7f6e9da3b9c0>
      f = <function Document.run_method.<locals>.fn at 0x7f6ea1ac0f40>
  File "apps/frappe/frappe/model/document.py", line 1304, in runner
    add_to_return_value(self, fn(self, *args, **kwargs))
      self = <CustomField: Bank Account-kosma_account_id>
      method = 'on_update'
      args = ()
      kwargs = {}
      add_to_return_value = <function Document.hook.<locals>.add_to_return_value at 0x7f6e9da68e00>
      fn = <function Document.run_method.<locals>.fn at 0x7f6ea1ac0f40>
      hooks = (<function clear_doctype_notifications at 0x7f6e9eab20c0>, <function process_workflow_actions at 0x7f6e9daefec0>, <function attach_files_to_document at 0x7f6ea0e41f80>, <function apply at 0x7f6e9db1ad40>, <function update_due_date at 0x7f6e9db1ade0>, <function apply_permissions_for_non_standard_user_type at 0x7f6e9db2c5e0>)
  File "apps/frappe/frappe/model/document.py", line 959, in fn
    return method_object(*args, **kwargs)
      self = <CustomField: Bank Account-kosma_account_id>
      args = ()
      kwargs = {}
      method_object = <bound method CustomField.on_update of <CustomField: Bank Account-kosma_account_id>>
      method = 'on_update'
  File "apps/frappe/frappe/custom/doctype/custom_field/custom_field.py", line 208, in on_update
    validate_fields_for_doctype(self.dt)
      self = <CustomField: Bank Account-kosma_account_id>
      validate_fields_for_doctype = <function validate_fields_for_doctype at 0x7f6e9db9bc40>
  File "apps/frappe/frappe/core/doctype/doctype/doctype.py", line 1200, in validate_fields_for_doctype
    validate_links_table_fieldnames(meta)
      doctype = 'Bank Account'
      meta = <Meta: Bank Account>
  File "apps/frappe/frappe/core/doctype/doctype/doctype.py", line 1150, in validate_links_table_fieldnames
    _test_connection_query(doctype=link.link_doctype, field=link.link_fieldname, idx=index)
      meta = <Meta: Bank Account>
      fieldnames = ('account_name', 'account', 'bank', 'account_type', 'account_subtype', 'column_break_7', 'disabled', 'is_default', 'is_company_account', 'company', 'section_break_11', 'party_type', 'column_break_14', 'party', 'account_details_section', 'iban', 'column_break_12', 'branch_code', 'bank_account_no', 'address_and_contact', 'address_html', 'column_break_13', 'contact_html', 'integration_details_section', 'integration_id', 'last_integration_date', 'column_break_27', 'mask', 'kosma_account_id')
      index = 4
      link = <DocTypeLink: vu8pjsoltm parent=Bank Account>
  File "apps/frappe/frappe/core/doctype/doctype/doctype.py", line 1195, in _test_connection_query
    frappe.throw(msg, InvalidFieldNameError)
      doctype = 'Payroll Entry'
      field = 'bank_account'
      idx = 4
      filters = {'bank_account': ''}
      msg = "Document Links Row #4: Invalid doctype or fieldname.<br>('DocType', 'Payroll Entry')"
  File "apps/frappe/frappe/__init__.py", line 652, in throw
    msgprint(
      msg = "Document Links Row #4: Invalid doctype or fieldname.<br>('DocType', 'Payroll Entry')"
      exc = <class 'frappe.core.doctype.doctype.doctype.InvalidFieldNameError'>
      title = None
      is_minimizable = False
      wide = False
      as_list = False
      primary_action = None
  File "apps/frappe/frappe/__init__.py", line 617, in msgprint
    _raise_exception()
      msg = "Document Links Row #4: Invalid doctype or fieldname.('DocType', 'Payroll Entry')"
      title = None
      raise_exception = <class 'frappe.core.doctype.doctype.doctype.InvalidFieldNameError'>
      as_table = False
      as_list = False
      indicator = 'red'
      alert = False
      primary_action = None
      is_minimizable = False
      wide = False
      realtime = False
      sys = <module 'sys' (built-in)>
      _raise_exception = <function msgprint.<locals>._raise_exception at 0x7f6e9da3b240>
      inspect = <module 'inspect' from '/usr/lib/python3.11/inspect.py'>
      out = {'message': "Document Links Row #4: Invalid doctype or fieldname.<br>('DocType', 'Payroll Entry')", 'title': 'Message', 'indicator': 'red', 'raise_exception': 1, '__frappe_exc_id': '7e99fe0923b68e2ae2852e79c8c3d1b7576d07f510e3d7a4f31da3f3'}
  File "apps/frappe/frappe/__init__.py", line 568, in _raise_exception
    raise exc
      exc = InvalidFieldNameError("Document Links Row #4: Invalid doctype or fieldname.('DocType', 'Payroll Entry')")
      inspect = <module 'inspect' from '/usr/lib/python3.11/inspect.py'>
      msg = "Document Links Row #4: Invalid doctype or fieldname.('DocType', 'Payroll Entry')"
      out = {'message': "Document Links Row #4: Invalid doctype or fieldname.<br>('DocType', 'Payroll Entry')", 'title': 'Message', 'indicator': 'red', 'raise_exception': 1, '__frappe_exc_id': '7e99fe0923b68e2ae2852e79c8c3d1b7576d07f510e3d7a4f31da3f3'}
      raise_exception = <class 'frappe.core.doctype.doctype.doctype.InvalidFieldNameError'>
frappe.core.doctype.doctype.doctype.InvalidFieldNameError: Document Links Row #4: Invalid doctype or fieldname.('DocType', 'Payroll Entry')
```